aten/src/ATen/native/cuda/linalg/BatchLinearAlgebra.cpp

```diff
@@ -1613,13 +1613,17 @@ static void lu_factor(const Tensor& input, const Tensor& pivots, const Tensor& i
   };
 
   const auto preferred_backend = at::globalContext().linalgPreferredBackend();
-#if defined(USE_LINALG_SOLVER) && !defined(USE_ROCM)
+#ifdef USE_LINALG_SOLVER
   const auto lu_factor_cusolver = [batch_size, m, n](const Tensor& input, const Tensor& pivots, const Tensor& infos, bool compute_pivots) {
     // In CUDA 10.2, lu_factor_looped_cusolver does not finish the computations when the input
     // matrix is exactly singular. The returned pivots contain garbage. This breaks linalg.det
     // Now, batched_cublas does not handle rectangular matrices, so we still dispatch to
     // looped_cusolver even if m != n.
+#ifdef USE_ROCM
+    bool looped_correct = true;
```

Comment:
```suggestion
    constexpr bool looped_correct = true;
```
To match the declaration used for the cusolver path.